src/test/java/com/amazon/ion/impl/EncodingDirectiveCompilationTest.java

```diff
@@ -1141,9 +1145,184 @@ public void taglessExpressionGroup(InputType inputType, StreamType streamType) t
         }
     }
 
+    private static void writeSymbolTableMacroInvocation(boolean isAppend, IonRawWriter_1_1 writer, String... symbols) {
+        writer.stepInEExp(isAppend ? SystemMacro.AddSymbols : SystemMacro.SetSymbols);
+        writer.stepInExpressionGroup(false);
+        for (String symbol : symbols) {
+            writer.writeString(symbol);
+        }
+        writer.stepOut();
+        writer.stepOut();
+    }
+
+    private static Map<String, Integer> writeSymbolTableSetMacroInvocation(IonRawWriter_1_1 writer, String... symbols) {
+        writeSymbolTableMacroInvocation(false, writer, symbols);
+        return makeSymbolsMap(FIRST_LOCAL_SYMBOL_ID, symbols);
+    }
+
+    private static void writeSymbolTableAppendMacroInvocation(IonRawWriter_1_1 writer, Map<String, Integer> existingSymbols, String... newSymbols) {
```

Comment:
The combination of "SymbolTableAppendMacro" got me confused. Maybe E-Expression would be a better choice here (and in the similar methods)?
```suggestion
    private static void writeSymbolTableAppendEExpression(IonRawWriter_1_1 writer, Map<String, Integer> existingSymbols, String... newSymbols) {
```

Reply:
Done.